edb/server/compiler/compiler.py

```diff
@@ -1822,17 +1836,133 @@ def _compile_ql_query(
 
     result_cardinality = enums.cardinality_from_ir_value(ir.cardinality)
 
+    # This low-hanging-fruit is temporary; persistent cache should cover all
+    # cacheable cases properly in future changes.
+    use_persistent_cache = (
+        ctx.persistent_cache and
+        cacheable and
+        not ctx.bootstrap_mode and
+        script_info is None
+    )
```

Comment:
I don't think this is worth your time to change everywhere now, and we aren't fully consistent, but the style in the compiler tends towards putting the `and` at the start of lines